pkg/kubelet/status/status_manager.go

```diff
@@ -136,11 +141,17 @@ type Manager interface {
 	// the provided podUIDs.
 	RemoveOrphanedStatuses(podUIDs map[types.UID]bool)
 
-	// GetPodResizeStatus returns cached PodStatus.Resize value
-	GetPodResizeStatus(podUID types.UID) v1.PodResizeStatus
+	// GetPodResizeStatus returns cached PodStatus Resize conditions value
+	GetPodResizeStatus(podUID types.UID) PodResizeStatus
+
+	// SetPodResizePendingCondition caches the last PodResizePending condition for the pod.
+	SetPodResizePendingCondition(podUID types.UID, resizePending *v1.PodCondition)
```

Comment:
Rather than passing in the full condition, what do you think about just passing in the values that are needed to generate the condition, and let the status manager handle the book-keeping? Something like

```go
SetPodResizePendingCondition(pod *v1.Pod, reason string, msg string)
```

Then the implementation of this method can handle setting everything else:
- type
- ObservedGeneration
- status=true
- lastProbeTime = when the method was called
- lastTransitionTime = now if the condition is new, previous value if the condition was already set and the reason is unchanged

Reply:
sgtm! updated